### ISSUE TITLE
Automating deployments to the beta site on update to master

### ISSUE THREAD

**Issue by jglovier** [HospitalRun/hospitalrun-frontend#388]
**Expected behavior:** When we merge pull requests to master, the system should deploy the latest to the beta site.

**Actual behavior:** We have to manually deploy to the beta site.

We can easily accomplish this with GitHub webhooks.

Some links for reference:
- https://developer.github.com/guides/automating-deployments-to-integrators/
- https://github.com/blog/2056-automating-code-deployment-with-github-and-azure
- https://github.com/blog/1241-deploying-at-github
- https://developer.github.com/v3/repos/deployments/

**Comment by ThomasWunderlich** [comment]
@jglovier Do you have a script you currently run to deploy? Or alternatively documented deployment steps? I went looking through the github repo and I wasn't sure what you're currently using for hosting or deployment of the beta site.

**Comment by jglovier** [comment]
@ThomasWunderlich it's pretty manual right now. I think @jkleinsc just logs into our host and `git pulls` basically LOL. 😄  cc @jkleinsc to confirm...

**Comment by ThomasWunderlich** [comment]
Well that at least confirms you're using a vm then and not docker. It makes the deployment process a bit easier

**Comment by jkleinsc** [comment]
@ThomasWunderlich @jglovier yup... right now I do the following:
1. Manual git pull on https://github.com/HospitalRun/hospitalrun-server
2. npm install
3. service hospitalrun restart

**Comment by jglovier** [comment]
Ideally we'd use something like [Heaven](https://github.com/atmos/heaven) so we can deploy via Hubot in Slack.

**Comment by parkr** [comment]
I would be interested in helping with this today. A v0.1 could replicate what @jkleinsc does above but via a cron job. That would be very simple to setup. Everything is public so it could be as simple as:

```bash
cat < > /etc/cron.d/update_server
MAILTO=cron@hospitalrun.io
0 * * * * root cd /data/hospitalrun-server && git pull origin master && npm install && service hospitalrun restart
```

A v1.0 could be:

- [ ] Stand up Heaven in Heroku. Give it a db & configure the apps needed to deploy.
- [ ] Configure webhook to hit Heaven. I believe it's just the Deployment event.
- [ ] Configure [hubot-deploy](https://github.com/atmos/hubot-deploy) in HospitalRun Hubot.
- [ ] ... anything else?

**Comment by parkr** [comment]
OK, after talking to @jglovier and @tangollama, it sounds like a general-purpose solution for deploying instances of hospitalrun-server would be better. For any instance that the HospitalRun team manages, it will be registered with heaven during the provisioning. Then, `.deploy hospitalrun-server to prod` would deploy via Heaven to all the hosted unites HospitalRun 

- [x] Setup a heaven Droplet. (Thanks @tangollama!)
- [x] Install Docker ([instructions used](https://docs.docker.com/engine/installation/linux/ubuntulinux/))
- [ ] Install Heaven via Docker ([instructions used](https://github.com/atmos/heaven/blob/e970e9c605bc134e0ec94f7fe3eb1b366d83dbaa/doc/installation.md#launch-using-docker))
- [ ] Configure Heaven to know about `hospitalrun-server`.
- [ ] Add [`hubot-deploy`](https://github.com/atmos/hubot-deploy) to HospitalRun's Hubot.
- [ ] Get `hubot-deploy` talking to Heaven.

**Comment by tangollama** [comment]
Note: @parkr is working on this.

**Comment by jglovier** [comment]
@parkr I started installing `hubot-deploy` on our existing [Slack hubot(https://github.com/HospitalRun/hospitalbot) when I realized you created the @hospitalrunbot user. Should I deprecate the `hospitalbot` repo in favor of that new bot user?

**Comment by tangollama** [comment]
@parkr @jglovier what do we need to do to finish this?

**Comment by tangollama** [comment]
@parkr @jglovier, consider this a very friendly bump. I'm happy to work some of this if that's feasible.

**Comment by jpeterson88** [comment]
Hey there, just bumping this to see where it currently stands @jglovier

**Comment by jpeterson88** [comment]
Sweet. Yeah I'd love to take a stab at this. I see there is a discussion of using Heaven. I'll start digging into documentation this weekend. Any additional info the team has would be much appreciated. Assuming I'll need credentials or whatever to where the site is hosted, who should I reach out to @jglovier ?

**Comment by jglovier** [comment]
> I see there is a discussion of using Heaven, are we set on that? Why not something like Kubernetes + Helm?

I'm open to other solutions, I just proposed Heaven because it's what we used internally at GitHub and I was familiar with it a bit. But I'm not actively working on this issue and not strongly opinionated as to implementation details, so definitely open to other solutions that make sense. 👍 

> Assuming I'll need credentials or whatever to where the site is hosted, who should I reach out to @jglovier ?

Good question. @jkleinsc or @tangollama would have credentials for those.
